src/main/java/org/gaul/s3proxy/ShardedBlobStore.java

```diff
@@ -0,0 +1,677 @@
+        }
+        // TODO: implement listing a sharded container
+        throw new UnsupportedOperationException("sharded bucket");
+    }
+
+    @Override
+    public PageSet<? extends StorageMetadata> list(
+            String container,
+            ListContainerOptions options) {
+        if (!this.buckets.containsKey(container)) {
+            return this.delegate().list(container, options);
+        }
+        // TODO: implement listing a sharded container
+        throw new UnsupportedOperationException("sharded bucket");
+    }
+
+    @Override
+    public boolean containerExists(String container) {
+        if (!this.buckets.containsKey(container)) {
+            return this.delegate().containerExists(container);
+        }
+        return true;
+    }
+
+    @Override
+    public ContainerAccess getContainerAccess(String container) {
+        if (!this.buckets.containsKey(container)) {
+            return this.delegate().getContainerAccess(container);
+        }
+        throw new UnsupportedOperationException("sharded bucket");
+    }
+
+    @Override
+    public void setContainerAccess(String container,
+                                   ContainerAccess containerAccess) {
+        if (!this.buckets.containsKey(container)) {
+            this.delegate().setContainerAccess(container, containerAccess);
+        }
+        throw new UnsupportedOperationException("sharded bucket");
+    }
+
+    @Override
+    public void clearContainer(String container) {
+        clearContainer(container, new ListContainerOptions());
+    }
+
+    @Override
+    public void clearContainer(String container, ListContainerOptions options) {
+        throw new UnsupportedOperationException("sharded bucket");
+    }
+
+    @Override
+    public void deleteContainer(String container) {
+        throw new UnsupportedOperationException("sharded bucket");
+    }
+
+    private boolean deleteShards(ShardedBucket bucket) {
+        ImmutableMap.Builder<Future<Boolean>, String> futuresMapBuilder =
+                new ImmutableMap.Builder<>();
+        Exception err = null;
+        ExecutorService executor = Executors.newFixedThreadPool(
+                Math.min(bucket.shards, MAX_SHARD_THREADS));
+        IntStream.range(0, bucket.shards).forEach(
+            n -> futuresMapBuilder.put(executor.submit(
+                () -> this.delegate().deleteContainerIfEmpty(
+                        ShardedBlobStore.getShardContainer(bucket, n))),
+                    ShardedBlobStore.getShardContainer(bucket, n))
+        );
+        executor.shutdown();
+        ImmutableMap<Future<Boolean>, String> futuresMap =
+                futuresMapBuilder.build();
+        boolean ret = true;
+        for (Future<Boolean> result : futuresMap.keySet()) {
+            String shardName = futuresMap.get(result);
+            try {
+                boolean status = result.get();
+                if (!status) {
+                    logger.warn("Failed to delete " + shardName);
```

Comment:
Prefer {} expansion:

```java
                     logger.warn("Failed to delete {}", shardName);
```

Reply:
Removed this logging entirely, as a consequence of re-raising the first exception.